### PR TITLE
feat(l10n): Allow to force language with a query parameter

### PR DESCRIPTION
**Pull request by come-nc**
* Resolves: # 

## Summary

This will be used by support application to always generate the support
 report in english. It may also be used to debug a language, or to
 temporarily show a page in an other language to take a screenshot or to
 share your screen with someone speaking another language.

## TODO

- [ ] Should that be a header instead?
- [ ] Where should that be documented?
- [ ] Can it cause any troubles anywhere? I don’t think so :thinking: 

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by AndyScherzinger — APPROVED** [review]

File: lib/private/L10N/Factory.php [hunk 1]
```diff
@@ -98,7 +98,7 @@ public function get($app, $lang = null, $locale = null) {
 				$lang = str_replace(['\0', '/', '\\', '..'], '', $lang);
 			}
 
-			$forceLang = $this->config->getSystemValue('force_language', false);
+			$forceLang = $this->request->getParam('forceLanguage') ?? $this->config->getSystemValue('force_language', false);
```
Comment (nickvergessen) [inline lib/private/L10N/Factory.php:101]: Also should overwrite in those?
- getLanguageFromRequest
- findLanguage
- findGenericLanguage

File: lib/private/L10N/Factory.php [hunk 2]
```diff
@@ -411,6 +411,10 @@ public function getUserLanguage(?IUser $user = null): string {
 				return $language;
 			}
 
+			if (($forcedLanguage = $this->request->getParam('forceLanguage')) !== null) {
```
Comment (nickvergessen) [inline lib/private/L10N/Factory.php:414]: In the other methods it takes priority over the config, should it also here?

Comment (come-nc) [reply lib/private/L10N/Factory.php:414]: I hesitated. force_language from configuration does takes precedence. But for forceLanguage query parameter it felt weird to take priority over user language when specifically asking for it.

But I’m not sure here, because if the calling code tries to get an L10N instance using that language it will still get the forced one, so I’m not sure whether it can be useful.

Basically the corner-case here is if I use ?forceLanguage=de and I share a file by email it may send the email in German regardless of the target language setting. But 1) either way is fine to me, it’s really a corner case and 2) it’s not really gonna do that any way because the request from frontend button clic will not have the query parameter.

So, tl;dr:
I’m fine with forcing language on first line of getUserLanguage as other methods do if you think it’s better.
If we do that, maybe we should even force it in getLanguages like the config setting is doing, making the config option and the query parameter consistent with each other.

Comment (nickvergessen) [reply lib/private/L10N/Factory.php:414]: lets do it and see where we end up.
Have to check that sending emails in target language doesn't fail, but I guess the amount of requests where we `?forceLanguage` will also be very limited.

**Review by nickvergessen — APPROVED** [review]
